cloud/disk_manager/internal/pkg/dataplane/url/vhd/image_map_reader.go

```diff
@@ -95,3 +138,70 @@ func (r *ImageMapReader) readHeader(ctx context.Context) error {
 
 	return r.header.validate()
 }
+
+func (r *ImageMapReader) readBAT(ctx context.Context) error {
+	r.bat = make([]uint32, r.header.MaxTableEntries)
+
+	return r.reader.ReadBinary(
+		ctx,
+		r.header.TableOffset,
+		r.getBATSizeInBytes(),
+		binary.BigEndian,
+		&r.bat,
+	)
+}
+
+func (r *ImageMapReader) getBATSizeInBytes() uint64 {
+	return uint64(r.header.MaxTableEntries) * 32 / 8
```

Comment:
BAT entry is 4 bytes long. Why * 32 / 8? And add a constant. The address space of the VHD seems to be 4 bytes unsigned int.